netflix/src/redux/api.ts

```diff
@@ -0,0 +1,50 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+export const filmApi = createApi({
+  reducerPath: "filmApi",
+  baseQuery: fetchBaseQuery({ baseUrl: "http://localhost:4000" }),
+  endpoints: (builder) => ({
+    getMovieById: builder.query({
+      query: (id) => `/movies/${id}`,
+    }),
+
+    getMoviesList: builder.query({
+      query: () => `/movies`,
+    }),
+
+    searchMovie: builder.query({
+      query: (searchQuery) => `/movies?search=${searchQuery}&searchBy=title`,
+    }),
+
+    filter: builder.query({
+      query: ({ genre, sortBy, sortOrder }) => {
```

Comment:
too complex...
better define default parameters and use JS built in API https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams
```
filter: builder.query({
      query: ({ genre, sortBy = 'vote_average', sortOrder = 'desc' }) => {
        const queryParams = new URLSearchParams();
        queryParams.append('sortBy', sortBy);
        queryParams.append('sortOrder', sortOrder);
        if (genre) {
          queryParams.append('filter', genre);
        }
        return `/movies${queryParams.toString()}`;
      },
    }),
```

Reply:
Overall building query string on the fly is bad practice, because HTTP protocol require encryption of special characters, which is not handled properly by simple String... While `URLSearchParams` will do all this work for you